src/json-filter.js

```diff
@@ -47,6 +47,17 @@ function jsonFilter(query, log) {
     if (Array.isArray(json.data)) {
       body = filter(json);
     } else {
+      if (!json[NAMES_KEY]) {
+        const msg = 'multisheet data invalid. missing ":names" property.';
+        log.info(msg);
+        return new Response('', {
+          status: 404,
```

Comment:
This should be a 502, not a 404.

> 502 Bad Gateway server error response code indicates that the server, while acting as a gateway or proxy, received an invalid response from the upstream server.

An invalid JSON file (not according to schema) would definitely classify as invalid response.

Reply:
@tripodsan do you want me to open a new issue?

Reply:
the upstream server didn't provide an invalid response, but the data is not expected.

also, there is nothing the client can do to remedy the error. since this service is providing a resource document to the client, I think a 404 is better.

Reply:
The client cannot do anything, so the fault is on the server-side, ergo 5xx not 4xx.

Out of all known 5xx errors, 502 seems to be the best fit to describe the scenario when the response is not `application/json+multisheet` but just `application/json+somethingelse`.

As a 404, it will get lost among the hundreds of client-induced errors (and ignored). As a 502, we know what to do and can fix the content bus that generated the bad file.

Reply:
> As a 404, it will get lost among the hundreds of client-induced errors (and ignored). As a 502, we know what to do and can fix the content bus that generated the bad file.

it will not be ignored, since the developer will miss the data.

Anyways, other valid options would be (in this case):
- best guess, and ignore the missing `:names` property
- or return an empty document